commands/qemu-build/buildimage.go

```diff
@@ -133,3 +136,43 @@ func buildImage(
 
 	return nil
 }
+
+// load vm.Machine from file without migration
+func newMachineFromFile(machineFile string) (*vm.Machine, error) {
+	// Read machine.json
+	machineData, err := ioext.BoundedReadFile(machineFile, 1024*1024)
+	if err == ioext.ErrFileTooBig {
+		return nil, runtime.NewMalformedPayloadError(
+			"The file 'machine.json' larger than 1MiB. JSON files must be small.")
```

Comment:
If you construct an image file that contains `machine.json` that is 10GiB large I might happily download and extract the image... but I won't try to load it into memory... As it would crash the worker..

Images are untrusted.